gen-json-schema-class.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+import yaml
+import os
+import subprocess
+import json
+import time
+from pathlib import Path
+from collections import OrderedDict
+import jsonref
+import synapseclient
+
+# ─── HELPER FUNCTIONS ─────────────────────────────────────────────────────────
+
+def run_capture(cmd, **kw):
+    print("⟳", " ".join(str(c) for c in cmd))
+    try:
+        result = subprocess.run(cmd, check=True, stdout=subprocess.PIPE, text=True, **kw)
+        return result.stdout
+    except subprocess.CalledProcessError as e:
+        print(f"❌ Command failed: {e}")
+        return None
+
+
+def remove_keys(obj, keys_to_remove):
+    """
+    Recursively remove any dict entries whose key is in keys_to_remove.
+    """
+    if isinstance(obj, dict):
+        for k in list(obj.keys()):
+            if k in keys_to_remove:
+                obj.pop(k)
+            else:
+                remove_keys(obj[k], keys_to_remove)
+    elif isinstance(obj, list):
+        for item in obj:
+            remove_keys(item, keys_to_remove)
+
+# ─── CONFIG ───────────────────────────────────────────────────────────────────
+SCHEMA_YAML = Path("dist/NF.yaml")
+HEADER_YAML = Path("header.yaml")
+BUILD_DIR   = Path("modules/Template")
+OUT_DIR     = Path("registered-json-schemas")
+
+BUILD_DIR.mkdir(parents=True, exist_ok=True)
+OUT_DIR.mkdir(parents=True, exist_ok=True)
+
+# ─── LOAD MASTER SCHEMA ───────────────────────────────────────────────────────
+master      = yaml.safe_load(SCHEMA_YAML.read_text(encoding="utf-8"))
+all_classes = master.get("classes", {})
+all_slots   = master.get("slots", {})
+all_enums   = master.get("enums", {})
+
+# ─── UTILITIES ────────────────────────────────────────────────────────────────
+def collect_hierarchy(cls_name):
+    hierarchy = []
+    cur = cls_name
+    while cur:
+        cdef = all_classes.get(cur)
+        if not cdef:
+            print(f"⚠️ Warning: class '{cur}' not found; skipping")
+            break
+        hierarchy.append((cur, cdef))
+        cur = cdef.get("is_a")
+    return hierarchy
+
+
+def slot_enums(sdef):
+    enums = set()
+    if "range" in sdef:
+        enums.add(sdef["range"])
+    if "any_of" in sdef:
+        enums.update(alt["range"] for alt in sdef["any_of"] if "range" in alt)
+    return enums
+
+# ─── STEP 1: GENERATE CLASS-SPECIFIC YAML ─────────────────────────────────────
```

Comment:
Hmm, so we don't actually need this code for class-specific YAML since LinkML can do this automatically with `gen-json-schema --top-class ..`, and they get kind of overwhelming under src/modules/Template